functions/Test-DbaServerName.ps1

```diff
@@ -0,0 +1,195 @@
+
+Returns ServerInstanceName, SqlServerName, IsEqual and RenameRequired for sqlserver2014a and sql2016.
+
+.EXAMPLE   
+Test-DbaServerName -SqlServer sqlserver2014a, sql2016 -Detailed
+
+Returns ServerInstanceName, SqlServerName, IsEqual and RenameRequired for sqlserver2014a and sql2016.
+	
+If a Rename is required, it will also show Updatable, and Reasons if the servername is not updatable.
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[PsCredential]$Credential,
+		[switch]$Detailed,
+		[switch]$NoWarning
+	)
+	
+	BEGIN
+	{
+		$collection = New-Object System.Collections.ArrayList
+	}
+	
+	PROCESS
+	{
+		$servercount++
+		
+		foreach ($servername in $SqlServer)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $servername -SqlCredential $Credential
+			}
+			catch
+			{
+				if ($servercount -eq 1)
+				{
+					throw $_
+				}
+				else
+				{
+					Write-Warning "Can't connect to $servername. Moving on."
+					Continue
+				}
+			}
+			
+			
+			if ($server.isClustered)
+			{
+				Write-Warning "$servername is a cluster. Renaming clusters is not supported by Microsoft."
+			}
+			
+			$sqlservername = $server.ConnectionContext.ExecuteScalar("select @@servername")
+			$instance = $server.InstanceName
+			
+			if ($instance.length -eq 0)
+			{
+				$serverinstancename = $server.NetName
+				$instance = "MSSQLSERVER"
+			}
+			else
+			{
+				$netname = $server.NetName
+				$serverinstancename = "$netname\$instance"
+			}
+			
+			$serverinfo = [PSCustomObject]@{
+				ServerInstanceName = $serverinstancename
+				SqlServerName = $sqlservername
+				IsEqual = $serverinstancename -eq $sqlservername
+				RenameRequired = $serverinstancename -ne $sqlservername
+			}
+			
+			if ($Detailed)
+			{
+				$reasons = @()
+				$servicename = "SQL Server Reporting Services ($instance)"
+				
+				try
+				{
+					$rs = Get-Service -ComputerName $server.ComputerNamePhysicalNetBIOS -DisplayName $servicename -ErrorAction SilentlyContinue
+				}
+				catch
+				{
+					if ($NoWarnings -eq $false)
+					{
+						Write-Warning "Can't contact $($server.ComputerNamePhysicalNetBIOS) using Get-Service. This means the script will not be able to automatically restart SQL services."
+					}
+				}
+				
+				if ($rs.count -gt 0)
+				{
+					$rstext = "Reporting Services must be stopped and updated"
+					$serverinfo | Add-Member -NotePropertyName Warnings -NotePropertyValue $rstext
+				}
+				
+				# check for mirroring
+				$mirroreddb = $server.Databases | Where-Object { $_.IsMirroringEnabled -eq $true }
+				
+				if ($mirroreddb.count -gt 0)
+				{
+					$dbs = $mirroreddb.name -join ", "
+					$reasons += "Databases are being mirrored: $dbs"
+				}
+				
+				# check for replication
+				$sql = "select name from sys.databases where is_published = 1 or is_subscribed =1 or is_distributor = 1"
```

Comment:
This T-SQL fails on SQL 2000, throws an error on screen. Had to add my own Write-Verbose to find out what was causing the error.